--- papi-1.0_beta/source/libpapi-common/uri.c.orig	2005-09-21 21:04:53.000000000 +0200
+++ papi-1.0_beta/source/libpapi-common/uri.c	2010-04-02 09:38:15.848988896 +0200
@@ -15,22 +15,6 @@
 #include <errno.h>
 #include "uri.h"
 
-static char *
-strndup(char *string, size_t length)
-{
-	char *result = NULL;
-
-	if (length > 0) {
-		length++;
-
-
-		if ((result = calloc(1, length)) != NULL)
-			(void) strlcat(result, string, length);
-	}
-
-	return (result);
-}
-
 
 /*
  * This will handle the following forms:
--- papi-1.0_beta/configure.in.orig	2006-03-12 07:15:33.000000000 +0100
+++ papi-1.0_beta/configure.in	2011-11-13 09:40:56.380634763 +0100
@@ -29,7 +29,7 @@
 AC_PROG_GCC_TRADITIONAL
 
 dnl Check for libraries
-AC_SEARCH_LIBS(rresvport,socket,,nsl)
+AC_SEARCH_LIBS(rresvport,socket nsl)
 AC_SEARCH_LIBS(rresvport_af,nsl socket,[AC_DEFINE(HAVE_RRESVPORT_AF, 1,
 	[define if you have rresvport_af])],)
 AC_SEARCH_LIBS(getipnodebyname,nsl socket,[AC_DEFINE(HAVE_GETIPNODEBYNAME, 1,
--- papi-1.0_beta/source/libpapi-lpd/lpd-port.c.orig	2005-09-21 21:22:22.000000000 +0200
+++ papi-1.0_beta/source/libpapi-lpd/lpd-port.c	2020-08-30 19:10:47.554210812 +0200
@@ -23,7 +23,6 @@
 #include <errno.h>
 #include <syslog.h>
 #include <values.h>
-#include <stropts.h>	/* for sendfd */
 #include <sys/uio.h>	/* for sendmsg stuff */
 #include <pwd.h>
 #include <sys/sendfile.h>
--- papi-1.0_beta/source/libpapi-lpd/lpd-misc.c.orig	2005-09-21 21:22:10.000000000 +0200
+++ papi-1.0_beta/source/libpapi-lpd/lpd-misc.c	2020-08-30 19:34:17.756571090 +0200
@@ -17,7 +17,6 @@
 #include <sys/socket.h>
 #include <errno.h>
 #include <wait.h>
-#include <stropts.h>
 #include <papi_impl.h>
 
 #include <config-site.h>
